plugins/omelasticsearch/omelasticsearch.c

```diff
@@ -2092,6 +2099,12 @@ CODESTARTnewActInst
 	if(pData->searchType == NULL)
 		pData->searchType = (uchar*) strdup("events");
 
+	/* if the search type is explicitly set to an empty string, omit it entirely */
+	if(!strcmp((char*)pData->searchType, "")) {
+		free(pData->searchType);
```

Comment:
self-reviewing this bit, is freeing the dummy value here correct? I'm not a 100% sure on the ownership semantics here.

Reply:
looks good to me